lib/tracking/trackProduct.js

```diff
@@ -1,43 +1,37 @@
-import track from "./track";
 import getProductTrackingData from "./utils/getProductTrackingData";
 import getVariantTrackingData from "./utils/getVariantTrackingData";
 
 /**
- * trackProduct HOC tracks a product
+ * trackProduct to tracks a product
  * @name trackProduct
- * @param {Object} options options to supply to tracking HOC
+ * @param {Object} payload data to be tracked
+ * @param {Object} router router next js
  * @returns {React.Component} - component
  */
-export default (options) => (
-  track(({ product, router }, state, functionArgs) => {
-    let data = {};
-    const { variant, optionId, action } = (functionArgs && functionArgs[0]) || [];
+export default (payload, router) => {
+  const { product, variant, optionId, action } = payload;
+  let data = {};
+  // If product data is provided as a prop, then process the data for tracking
+  if (product) {
+    data = {
+      action,
+      ...getProductTrackingData(product, router)
+    };
 
-    // If product data is provided as a prop, then process the data for tracking
-    if (product) {
+    // Add variant data if available
+    if (variant) {
       data = {
-        action,
-        ...getProductTrackingData(product)
+        ...data,
+        ...getVariantTrackingData({
+          product,
+          router,
+          variant, // Object representing a variant. (Required)
```

Comment:
Is this covered in JSDoc somewhere so the editor will pull it up automatically using IntelliSense versus relying on comments?